build/webcomponents/js/field-category/field-category.js

```diff
@@ -0,0 +1,35 @@
+customElements.define('joomla-field-category', class extends HTMLElement {
+  constructor() {
+    super();
+
+    this.element = '';
+    Joomla.loadingLayer('load', document.body);
+  }
+
+  connectedCallback() {
+    // Check if custom fields are enabled
+    if (this.getAttribute('custom-fields-enabled') !== 'true') {
+      return;
+    }
+
+    this.element = this.querySelector('select');
+    this.categoryHasChanged = this.categoryHasChanged.bind(this);
+
+    if (!this.element.value !== this.getAttribute('custom-fields-cat-id')) {
+      this.element.value = this.getAttribute('custom-fields-cat-id');
+    }
+
+    this.element.addEventListener('change', this.categoryHasChanged);
+  }
+
+  categoryHasChanged() {
+    if (this.element.value === parseInt(this.element.parentNode.getAttribute('custom-fields-cat-id'), 10)) {
+      return;
+    }
+
+    Joomla.loadingLayer('show', document.body);
+
+    document.querySelector('input[name=task]').value = `${this.element.parentNode.getAttribute('custom-fields-section')}.reload`;
+    this.element.form.submit();
```

Comment:
That seems too tightly coupled to a specific use case... 
Do is always have a task? Do we alway want to submit? Is it always ${section}.reload... 

Much like in fedirs PR for the joomla-toolbar-button, i think i have a little different idea of the usage of custom elements in joomla.

Reply:
Don’t forget this was the code already there. Can it be done differently , sure but let’s take small steps or each task could become a never ending story

Reply:
Yes. You are absolutely right. But... (sry!) now we put it in a custom element. 

For me custom elements are basic, reusable elements of the joomla ui (like buttons, cards, icons, dropdowns, modals, progress, table, panels, breadcrumb, pagination... whatever)...

What we are doing here (and in other PRs) is using custom elements as a module/library pattern to solve very special use cases. Thats what im struggling with. 

We bind javascript behaviour to an html element, but we limit the usage to one special use case and we are not using the lifecycle callbacks. 

So at least in the current implementation, i have problems to see the reason why this should be a custom element instead of a simple script? Perhaps im missing something here?

Reply:
Ok, here are some questions:
- most of the custom elements we did so far **are** parts of a Joomla field, basically the UI of that field
- Fields are good to use CE as they do leverage the Browser Event system, which means that the subform will work by default with these components without implementing extra events.
- Fields implemented with CE **should** be decoupled from Boostrap meaning they're css framework agnostic.
- Down the road these should be converted to full blown WebComponents so we do have true encapsulation